src/test_workflow/benchmark_test/benchmark_test_suite.py

```diff
@@ -0,0 +1,54 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import subprocess
+from typing import Any
+
+from test_workflow.benchmark_test.benchmark_args import BenchmarkArgs
+
+
+class BenchmarkTestSuite:
+    endpoint: str
+    security: bool
+    current_workspace: str
+    args: BenchmarkArgs
+    command: str
+
+    """
+    Represents a performance test suite. This class runs rally test on the deployed cluster with the provided IP.
+    """
+    def __init__(
+            self,
+            endpoint: Any,
+            security: bool,
+            args: BenchmarkArgs,
+    ) -> None:
+        self.endpoint = endpoint
+        self.security = security
+        self.args = args
+        # Pass the cluster endpoints with -t for multi-cluster use cases(e.g. cross-cluster-replication)
+        if args.benchmark_config:
+            base_command = f"docker run -v {args.benchmark_config}:/opensearch-benchmark/.benchmark/benchmark.ini"
+        else:
+            base_command = "docker run"
+
+        self.command = (
+            f"{base_command} opensearchproject/opensearch-benchmark:latest execute_test "
+            f"--workload={self.args.workload} --test-mode --pipeline=benchmark-only --target-hosts={endpoint}"
+        )
```

Comment:
Looks messy, and uses an unnecessary local `base_command` variable.

```python
self.command = "docker run"
if args.benchmark_config:
   self.command += " -v ..."
self.command += "opensearchproject/opensearch-benchmark:latest execute_test:
...
```

Reply:
Thanks, done!